docs/1.guide/5.routing.md

```diff
@@ -0,0 +1,108 @@
+---
+icon: solar:route-linear
+---
+
+## Registering Routes
+
+H3 app instance has a built in router system to register routes matching patterns with parameter or specific HTTP method.
+
+::tip
+Router is powered by [🌳 rou3](https://github.com/unjs/rou3), an ultra fast and tiny route matcher engine.
+::
+
+In the example below, we register a route requests to the `/hello` endpoint with HTTP GET method:
+
+```js
+// Method 1: using app.[method]
+app.get("/hello", () => "Hello world!");
+
+// Method 2: using app.on(method)
+app.on("GET", "/hello", () => "Hello world!");
+```
+
+You can register multiple event handlers for the same route with different methods:
+
+```js
+app
+  .get("/hello", () => "GET Hello world!")
+  .post("/hello", () => "POST Hello world!");
+```
+
+You can also use `app.all` method to register a route accepting any HTTP method:
+
+```js
+app.all("/hello", (event) => `This is a ${event.req.method} request!`);
+```
+
+### Route Params
+
+You can define parameters in your routes using `:` prefix:
+
+```js
+app.get("/hello/:name", (event) => {
+  return `Hello ${event.context.params.name}!`;
+});
+```
+
+In this example, the `name` parameter will be available in the `event.context.params` object.
+
+If you send a request to `/hello/world`, the event handler will respond `Hello world!`.
+
+Instead of named params, you can use `*` for unnamed and **optional** params:
+
+```js
+app.get("/hello/*", (event) => `Hello!`);
+```
+
+### Catch-all param
+
+Adding `/hello/:name` route will match `/hello/world` or `/hello/123`. But it will not match `/hello/foo/bar`.
+
+If you need to match multiple levels of sub routes, you can use `**` prefix:
+
+```js
+app.get("/hello/**", (event) => `Hello ${event.context.params._}!`);
+```
+
+This will match `/hello`, `/hello/world`, `/hello/123`, `/hello/world/123`, etc.
+
+> [!NOTE]
+> Param `_` will store the full wildcard content as a single string.
+
+## Middleware
+
+You can register [event handlers](/guide/event-handler) to app instance using the `app.use`:
+
+```js
+app.use("/hello", () => "Hello world!");
+```
+
+This will register the event handler to the app instance and will be called for every request starting with the prefix `/hello`. This means that the event handler will be called for `/hello`, `/hello/world`, `/hello/123`, etc.
```

Comment:
If I'm correct, what will be called is the event handler, won't it?
Grammatically the subject is "This" (calling `app.use`) in this sentence.

```suggestion
This will register the event handler to the app instance, and it will be called for every request starting with the prefix `/hello` (i.e. `/hello`, `/hello/world`, `/hello/123`, etc.).
```